### ISSUE TITLE
Unable to retrieve child resources -> This request is not authorized to perform this operation

### ISSUE THREAD

**Issue by AsiaKosim** [microsoft/AzureStorageExplorer#3668]
Error when clicking on Blob Containers -> Unable to retrieve child resources -> This request is not authorized to perform this operation

![image](https://user-images.githubusercontent.com/61511941/95737720-713f6e00-0c88-11eb-8231-9564ceafddb2.png)

![image](https://user-images.githubusercontent.com/61511941/95737681-5e2c9e00-0c88-11eb-9794-cacedcd3844e.png)

**Comment by StargazerNC** [comment]
Here, when I try to expand the tree, I get this:

Unable to retrieve child resources.

Details:
["Error:Unable to retrieve authentication token."]

**Comment by MRayermannMSFT** [comment]
@AsiaKosim that error messages means you don't have the right permissions needed to access whatever it is you are trying to access. How are you connecting to your resources? Sign in and then via subscription? Attach with SAS? Attach with Azure AD? Attach with connection string?

@StargazerNC can you go to Settings -> Services -> Disks, uncheck the checkbox and then restart? That should unblock you (assuming you don't need to work with disks).

**Comment by mutjay** [comment]
Hi I am connecting from Azure VM within the same Virtual Network. Do I need to apply any changes on the VM.

**Comment by StargazerNC** [comment]
@MRayermannMSFT Actually, after a while I just closed and reopened Azure Storage Explorer,  and magically it all worked as always have been. It's worth noting that yesterday, around 4 pm (GMT) all kinds of strange stuff was happening on our Azure services (Functions, Databases, DevOps...), so I'll just blame that. 🤣

**Comment by AsiaKosim** [comment]
@MRayermannMSFT I was connecting via subscription based on my Azure account. So I would assume it will automatically show all the resources to which I have access. So, after successful authentication, it shown the resources tree correctly (as on the screen shot), but I get the error on the lowest level, when I try to open the Blob:
![image](https://user-images.githubusercontent.com/61511941/96427550-fb468400-11fe-11eb-8fd9-185c290d758e.png)

I was also trying to connect manually  based on storage account name and the key, but result was the same. 

And I definitely have the access to the resources via azure.portal:

![image](https://user-images.githubusercontent.com/61511941/96428311-ea4a4280-11ff-11eb-99fc-de560c916cd0.png)

**Comment by MRayermannMSFT** [comment]
@AsiaKosim are you running storage explorer on a VM which is in a VNet?

 @mutjay have you enabled storage firewall? If so, then you will need to add the VNet as an allowed network to the firewall?

**Comment by MRayermannMSFT** [comment]
Closing due to lack of response. Please open a new issue if you're still having problems.
